app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -0,0 +1,680 @@
+
+
+                // === Grid Resizing ===
+
+                eval scene.frp.shape ((shape) model.refit_grid_to_shape(shape));
+                eval scene.frp.shape ((shape) model.refit_entries_to_shape(shape));
+
+
+                // === Entries Models ===
+
+                // FIXME [NP]: How do we stop the grid from calling `model_for_entry` on the
+                // header row? It should be handled via the later extension to the network.
+                projects_table.model_for_entry <+
+                    projects_table.model_for_entry_needed.filter_map(f!((input) model.model_for_entry((*input).into()).map(|(position, entry_model)| {
+                        let (row, col) = position.into();
+                        (row, col, entry_model)
+                    })));
+            }
+
+            configure_projects_table_grid(projects_table, &model);
+            configure_projects_table_header(projects_table, network);
+            configure_projects_table_scroll_parameters(projects_table);
+
+            app.display.add_child(root);
+
+            populate_table_with_data(input);
+
+
+            Self { frp, model }
+        }
+    }
+
+    fn configure_projects_table_grid(projects_table: &ProjectsTable, model: &Model) {
+        let entry_size = Vector2(ENTRY_WIDTH, ENTRY_HEIGHT);
+        projects_table.set_entries_size(entry_size);
+        let params = ensogl_grid_view::simple::EntryParams {
+            // FIXME [NP]: Turn these into StyleWatchFrp values or at the very least constants.
+            bg_color: color::Lcha::transparent(),
+            bg_margin: 1.0,
+            // TODO [NP]: Apply the hover color to the whole row, not just the selected entry.
+            hover_color: color::Lcha::from(color::Rgba(
+                62f32 / u8::MAX as f32,
+                81f32 / u8::MAX as f32,
+                95f32 / u8::MAX as f32,
+                0.05,
+            )),
+            selection_color: color::Lcha::from(color::Rgba(1.0, 0.0, 0.0, 1.0)),
+            ..default()
+        };
+        projects_table.set_entries_params(params);
+        let row = model.rows();
+        let col = Columns::LEN;
+        projects_table.reset_entries(row, col);
+        projects_table.focus();
+    }
+
+    fn configure_projects_table_header(
+        projects_table: &ProjectsTable,
+        network: &enso_frp::Network,
+    ) {
+        let header_frp = projects_table.header_frp();
+        frp::extend! { network
+            requested_section <- header_frp.section_info_needed.map(|&(row, col)| {
+                let sections_size = 2 + col;
+                let section_start = row - (row % sections_size);
+                let section_end = section_start + sections_size;
+                let position = (section_start, col).into();
+                let model = header_entry_model(position);
+                (section_start..section_end, col, model)
+            });
+            header_frp.section_info <+ requested_section;
```

Comment:
@farmaazon this code comes from one of the existing header view examples, but I'm having some trouble understanding what it's doing. What is the logic here supposed to be doing?

Reply:
Generally, the grid view wants to know what to display as a header of the top of the viewport in the current column. Therefore, it asks about information about the section the topmost visible entry is in.

`section_info_needed` endpoint is exactly that: a question about the section the given entry is in. 

IT also assumes that there may be more than one section, therefore it needs to know the span of rows belonging to the returned section (so it knows when to ask about another one).

So, the implementation copied from the example does not make much sense, of course, because you have only one section, so the range should be always `1..=self.rows()`.

Reply:
Perfect, thank you :)